rosbag2_cpp/src/rosbag2_cpp/writer.cpp

```diff
@@ -182,10 +183,21 @@ void Writer::write(
   const std::string & topic_name,
   const std::string & type_name,
   const rclcpp::Time & time)
+{
+  write(message, topic_name, type_name, time, time);
+}
+
+void Writer::write(
+  std::shared_ptr<const rclcpp::SerializedMessage> message,
+  const std::string & topic_name,
+  const std::string & type_name,
+  const rclcpp::Time & recv_time,
+  const rclcpp::Time & send_time)
 {
   auto serialized_bag_message = std::make_shared<rosbag2_storage::SerializedBagMessage>();
   serialized_bag_message->topic_name = topic_name;
-  serialized_bag_message->time_stamp = time.nanoseconds();
+  serialized_bag_message->receive_time_stamp = recv_time.nanoseconds();
+  serialized_bag_message->publish_time_stamp = send_time.nanoseconds();
```

Comment:
If it will be `rcutils_time_point_value_t` we will not need to use `nanoseconds()` getter
```suggestion
  serialized_bag_message->receive_time_stamp = recv_timestamp;
  serialized_bag_message->publish_time_stamp = send_timestamp;
```